### DISCUSSION TITLE
Dynamic Filtering Using Lookup

### DISCUSSION THREAD

**Discussion by llay03** [khoait/DCE.PCF#39] [General] [answered]
Awesome tool Khoa!

Is it possible to have dynamic templating using a lookup instead of text? 
Eg. Company Name equals {{parentaccountid}}

**Comment by khoait** [comment]
Hi @llay03. Yes, you can insert the lookup field id or name using `{{ lookup_field.id }}` or `{{ lookup_field.name }}`. Make sure the lookup field is available on your form.

Cheers!

**Reply by crmvet** [reply]
> Hi @llay03. Yes, you can insert the lookup field id or name using `{{ lookup_field.id }}` or `{{ lookup_field.name }}`. Make sure the lookup field is available on your form.
> 
> Cheers!
Amazing PCF!
{{ lookup_field.name }} works very well if put into Contains operator on view filter.
{{ lookup_field.id }} can not be saved into Equals operator and does not work if put into Contains operator..
![image](https://github.com/khoait/DCE.PCF/assets/81534547/f815799e-2a81-4435-9221-57213e302cdf)

Could you please clarify or give an example how to filter basing on the record GUID of lookup, aka _value. Where {{ lookup_field.id }} should be placed in view filter? what operator?

**Reply by khoait** [reply]
hi @crmvet, filtering a lookup field with ID is tricky as it is validated in the filter editor. It is a limitation for us I guess. 

However we can workaround this by creating a dummy text field, and use a plugin/flow to populate the customer id as text to the dummy field, then use the dummy field to filter. This looks ugly but I can't find a better way for now.

**Reply by crmvet** [reply]
Makes sense! I just was assuming I'm missing the way how it was intended to work with ID as the documentation was highlighting usage of lookup nave and id.
But I think overall the feature to consume custom XML would be much more valuable and would resolve any potential limitations with filtering, including this one with lookup id.

**Reply by khoait** [reply, marked as answer]
@llay03 @crmvet it is now possible to pass in a custom FetchXML using Dynamic Lookup View Configuration from release 1.0.7.0. so you're lookup filter won't be restricted by the validation in the UI anymore. please see https://github.com/khoait/DCE.PCF/wiki/PolyLookup#lookup-view for more details
